python/pyspark/sql/utils.py

```diff
@@ -399,11 +399,13 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
         if is_remote() and "PYSPARK_NO_NAMESPACE_SHARE" not in os.environ:
             from pyspark.sql.connect.window import Window as ConnectWindow
 
-            return getattr(ConnectWindow, f.__name__)(*args, **kwargs)
+            if isinstance(args[0], ConnectWindow):
```

Comment:
not sure whether need it, since all `Window` methods are static

Reply:
LGTM if it works

Reply:
Thank you for pointing it out! You are right. I removed the unreachable exception and adjusted the docstring.